packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js

```diff
@@ -16,30 +16,72 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
+import React, {useCallback} from 'react';
 import {Button} from '@axelor/aos-mobile-ui';
-import {useTranslator, useSelector} from '@axelor/aos-mobile-core';
+import {useTranslator, useSelector, useDispatch} from '@axelor/aos-mobile-core';
 import {Expense} from '../../../types';
+import {sendExpense, validateExpense} from '../../../features/expenseSlice';
 
-const ExpenseDetailsValidationButton = ({
-  expense,
-  onValidate = () => {},
-  onSend = () => {},
-}) => {
+const ExpenseDetailsValidationButton = ({expense, mode}) => {
   const I18n = useTranslator();
+  const dispatch = useDispatch();
 
   const {user} = useSelector(state => state.user);
 
+  const sendExpenseAPI = useCallback(
+    (expenseId, version) => {
+      dispatch(
+        sendExpense({
+          expenseId: expenseId,
+          version: version,
+          userId: user?.id,
+          onExpense: true,
+        }),
+      );
+    },
+    [dispatch, user],
+  );
+
+  const validateExpenseAPI = useCallback(
+    (expenseId, version) => {
+      dispatch(
+        validateExpense({
+          expenseId: expenseId,
+          version: version,
+          userId: user?.id,
+          onExpense: true,
+          user: user,
+          mode: mode,
+        }),
+      );
+    },
+    [dispatch, mode, user],
+  );
```

Comment:
When coming from this screen, onExpense is always true , so user & mode are never used, please remove them